### DISCUSSION TITLE
Publish message when a list/stream is empty?

### DISCUSSION THREAD

**Discussion by Dronakurl** [ag2ai/faststream#1611] [Q&A] [answered]
I am using Redis. I want my faststream application to publish a certain message once another list (or stream).
Is there anything like this in faststream? 
Something like this:

```python
@broker.publisher("triggermeonempty")
@broker.subscriber(stream="checkme", trigger_on_empty=True)
def run_me(msg):
     return "checkme is empty, do something about it"
```

Or maybe is it possible to get the length of the list within the subscriber?

**Comment by Lancetnik** [comment, marked as answer]
@Dronakurl sorry for answering so late! Unfortunatelly, we have no such functional, but you can get access to raw Redis client inside your subscriber and make anything you want with it:

```python
from faststream.redis.annotations import Redis

@broker.subscriber(...)
async def handler(msg, redis: Redis):
    list_len: int = await redis.llen("listname")
```
